lib/elixir/lib/uri.ex

```diff
@@ -626,6 +626,14 @@ defmodule URI do
 end
 
 defimpl String.Chars, for: URI do
+  def to_string(%{host: host, authority: authority, path: path} = uri)
+      when (host != nil or authority != nil) and is_binary(path) and
+             binary_part(path, 0, 1) != "/" do
```

Comment:
I think we need to check not only for nil, but for empty strings, since when we build the struct, we nillify the empty strings.

this is for another PR, which I can take care of, since I tried to tackle it and it needs a bit of revamping if we want to fully cover it. just let me know if you are interested,as I have one more question